python/lsst/ap/association/packageAlerts.py

```diff
@@ -165,6 +177,80 @@ def createDiaSourceBBox(self, bboxSize):
             bbox = geom.Extent2I(bboxSize, bboxSize)
         return bbox
 
+    def createCcdDataCutout(self, cutout, sphPoint, photoCalib):
+        """Convert a cutout into a calibrate CCDData image.
+
+        Parameters
+        ----------
+        cutout : `lsst.afw.image.ExposureF`
+            Cutout to convert.
+        sphPoint : `lsst.geom.SpherePoint`
+            Center point of DiaSource on the sky.
+        photoCalib : `lsst.afw.image.PhotoCalib`
+            Calibrate object of the image the cutout is cut from.
+
+        Returns
+        -------
+        ccdData : `astropy.nddata.CCDData`
+            CCDData object storing the calibrate information from the input
+            difference or template image.
+        """
+        cutOutMinX = cutout.getBBox().minX - 1
+        cutOutMinY = cutout.getBBox().minY - 1
+        center = cutout.getWcs().skyToPixel(sphPoint)
+        calibCutout = photoCalib.calibrateImage(cutout.getMaskedImage())
+
+        cutoutWcs = wcs.WCS(naxis=2)
+        cutoutWcs.array_shape = (cutout.getBBox().getWidth(),
+                                 cutout.getBBox().getWidth())
+        cutoutWcs.wcs.crpix = [center.x - cutOutMinX, center.y - cutOutMinY]
+        cutoutWcs.wcs.crval = [sphPoint.getRa().asDegrees(),
+                               sphPoint.getDec().asDegrees()]
+        cutoutWcs.wcs.cd = self.makeLocalTransformMatrix(cutout.getWcs(),
+                                                         center,
+                                                         sphPoint)
+
+        return CCDData(
+            data=calibCutout.getImage().array,
+            uncertainty=VarianceUncertainty(calibCutout.getVariance().array),
+            flags=calibCutout.getMask().array,
+            wcs=cutoutWcs,
+            meta={"cutMinX": cutOutMinX,
+                  "cutMinY": cutOutMinY},
+            unit=u.nJy)
+
+    def makeLocalTransformMatrix(self, wcs, center, skyCenter):
+        """Create a local, linear approximation of the wcs transformation
+        matrix.
+
+        The approximation is created as if the center is at RA=0, DEC=0. All
+        comparing x,y coordinate are relative to the position of center. Matrix
+        is initially calculated with units arcseconds and then converted to
+        radians. This yields higher precision results due to quirks in AST.
+
+        Parameters
+        ----------
+        wcs : `lsst.afw.geom.SkyWcs`
+            Wcs to approximate
+        center : `lsst.geom.Point2D`
+            Point at which to evaluate the LocalWcs.
+        skyCenter : `lsst.geom.SpherePoint`
+            Point on sky to approximate the Wcs.
+
+        Returns
+        -------
+        localMatrix : `numpy.ndarray`
+            Matrix representation the local wcs approximation with units
+            degrees.
+        """
+        localGnomonicWcs = afwGeom.makeSkyWcs(
+            center, skyCenter, [[self._scale, 0], [0, self._scale]])
```

Comment:
I think this would be more readable if you first defined a variable such as `cdMatrix = [[self._scale, 0], [0, self._scale]]` and then used that here.